core/discovery.go

```diff
@@ -19,9 +19,48 @@ under the License.
 
 package core
 
-import "github.com/spf13/viper"
+import (
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
 
-// GetRootNode place holder function for providing a boostrap address for a Validating Peer.
-func GetRootNode() (string, error) {
-	return viper.GetString("peer.discovery.rootnode"), nil
+	"github.com/spf13/viper"
+
+	
+)
+
+// StaticDiscovery is an implementation of Discovery
+type StaticDiscovery struct {
+	rootNodes   []string
+	random      *rand.Rand
+	isValidator bool
+}
+
+// NewStaticDiscovery is a constructor of a Discovery implementation
+func NewStaticDiscovery(isValidator bool) *StaticDiscovery {
+	staticDisc := StaticDiscovery{}
+	staticDisc.rootNodes = strings.Split(viper.GetString("peer.discovery.rootnode"), ",")
+	staticDisc.random = rand.New(rand.NewSource(time.Now().Unix()))
+	staticDisc.isValidator = isValidator
+	return &staticDisc
+}
+
+// GetRootNode returns the first root node defined in peer.discovery.rootnode config
+// for validating peers, and a random one for non validating peers.
+func (sd *StaticDiscovery) GetRootNode() (string, error) {
```

Comment:
Why is the behavior different for validating peers?  Why not always return a random node?